### PR TITLE
some MathML operator dictionary fixes and tests

### PR DESCRIPTION
**Pull request by xworld21**
I noticed some typos in the MathML operator dictionary code and fixed them.

To be sure, I added a couple of artificial tests in t/post/simplemath.xml to verify that the code is working as intended (e.g. now `&&` is correctly recognised as binary operator, so it does not get lspace/rspace attributes).

### CODE REVIEW

File: lib/LaTeXML/Post/MathML/OperatorDictionary.pm [hunk 1]
```diff
@@ -116,8 +116,8 @@ sub lookup_category {
     #   then replace Content with the Unicode character "U+0320
     #   plus the index of Content in Operators_2_ascii_chars"
     #   and move to step 3.
-    elsif (my ($p) = grep { $$Operators_2_ascii_chars[$_] ? ($_) : (); } 0 .. $#$Operators_2_ascii_chars) {
-      $code1 += $p;
+    elsif (my $p = $$Operators_2_ascii_chars{$content}) {
+      $code1   = 0x320 + $p;
```
Comment (dginev) [inline lib/LaTeXML/Post/MathML/OperatorDictionary.pm:120]: Subtle edge case detail: 

```perl
elsif (my ($p) = (0)) {...}
``` 
from the previous code would have evaluated as true. 

But the hash lookup approach would be false for the 0 index. Probably gfood to add a test for `!!`.

Comment (dginev) [reply lib/LaTeXML/Post/MathML/OperatorDictionary.pm:120]: Can we just hardcode the exact final character codes in `$Operators_2_ascii_chars` and avoid the runtime arithmetic? Should be safer (and won't have 0 values).

**Review by dginev — CHANGES_REQUESTED** [review]

File: lib/LaTeXML/Post/MathML/OperatorDictionary.pm (hunk 1 above)
Comment (xworld21) [reply lib/LaTeXML/Post/MathML/OperatorDictionary.pm]: Good catch! Now harcoded directly in the hash.

**Review by dginev — APPROVED** [review]
I think this PR indeed follows MathML Core.

It is not easy to penetrate which of the details are actually relevant and how to test them - I am not sure the `!!` test I suggested actually reveals anything of interest, as it generates a simple `<mo>` without attributes. But the code looks safe now to my reading ✅

**Comment by xworld21** [comment]
> I am not sure the `!!` test I suggested actually reveals anything of interest, as it generates a simple `<mo>` without attributes

Actually, the lack of attributes is the best signal that the dictionary is working as intended. It's recognising that MathML thinks of `!!` as postfix, so there is no need for `lspace=0`. If you test with LaTeXML 0.8.8, it will behave correctly with `!!` by accident (it looks up `!`, which is also postfix), but it will do the wrong thing with `&&` (it tries `(` which has 'Default' category when infix, so it adds redundant lspace/rspace to make it a binary operator).

This is still fragile in so far as LaTeXML does *not* compute the infix/postfix/prefix form in the way MathML and could omit lspace/rspace in the wrong situation. If I find a way to trigger a bug there, I'll file an issue with the example.

**Review by brucemiller — APPROVED** [review]
seems to be reasonable and following spec, as best as I can understand. Thanks!
